src/SystemView.cpp

```diff
@@ -778,117 +863,213 @@ void SystemView::Update()
 
 	m_rot_x_to = Clamp(m_rot_x_to, -80.0f, 80.0f);
 
-	if (m_shipDrawing != OFF) {
-		RefreshShips();
-		// if we are attached to the ship, check if we not deleted it in the previous frame
-		if (m_viewedObject.type != Projectable::NONE && m_viewedObject.base == Projectable::SHIP) {
-			auto bs = m_game->GetSpace()->GetBodies();
-			if (std::find(bs.begin(), bs.end(), m_viewedObject.ref.body) == bs.end())
-				ResetViewpoint();
-		}
-		if (m_selectedObject.type != Projectable::NONE && m_selectedObject.base == Projectable::SHIP) {
-			auto bs = m_game->GetSpace()->GetBodies();
-			if (std::find(bs.begin(), bs.end(), m_selectedObject.ref.body) == bs.end())
-				m_selectedObject.type = Projectable::NONE;
-		}
-	}
-}
-
-void SystemView::MouseWheel(bool up)
-{
-	if (this != Pi::GetView())
-		return;
+	int scroll = inputMgr->GetMouseWheel();
 
-	if (m_displayMode == Mode::Orrery) {
-		if (!up)
-			m_zoomTo *= 1 / ((ZOOM_OUT_SPEED - 1) * WHEEL_SENSITIVITY + 1) / Pi::GetMoveSpeedShiftModifier();
+	if (scroll && m_displayMode == SystemView::Mode::Orrery) {
+		if (scroll < 0)
+			m_zoomTo *= 1 / ((ZOOM_OUT_SPEED - 1) * WHEEL_SENSITIVITY * speedMod + 1);
 		else
-			m_zoomTo *= ((ZOOM_IN_SPEED - 1) * WHEEL_SENSITIVITY + 1) * Pi::GetMoveSpeedShiftModifier();
-	} else {
-		m_atlasZoomTo -= (up ? 1.0 : -1.0) * ATLAS_SCROLL_SENS;
+			m_zoomTo *= ((ZOOM_IN_SPEED - 1) * WHEEL_SENSITIVITY * speedMod + 1);
+	} else if (scroll) {
+		m_atlasZoomTo -= (scroll > 0 ? 1.0 : -1.0) * ATLAS_SCROLL_SENS;
 	}
 }
 
-void SystemView::RefreshShips(void)
+void SystemMapViewport::Update(float ft)
 {
-	m_contacts.clear();
-	auto bs = m_game->GetSpace()->GetBodies();
-	for (auto s = bs.begin(); s != bs.end(); s++) {
-		if ((*s) != Pi::player &&
-			(*s)->GetType() == ObjectType::SHIP) {
+	m_objectTracks.clear();
+	m_orbitTracks.clear();
+	m_projected.clear();
 
-			const auto c = static_cast<Ship *>(*s);
-			m_contacts.push_back(std::make_pair(c, c->ComputeOrbit()));
-		}
+	// TODO: add "true" lower/upper bounds to m_zoomTo / m_zoom
```

Comment:
should this TODO be fixed?

Reply:
This is a low-priority old TODO, will not be fixed in this PR.